controllers/assignments.py

```diff
@@ -342,6 +342,22 @@ def calculate_totals():
     else:
         return json.dumps({'success': False, 'message': "Select an assignment before trying to calculate totals."})
 
+@auth.requires_login()
+def student_autograde():
+    """
+    This is a safe endpoint that students can call from the assignment page
+    to get a preliminary grade on their assignment.
+    """
+    assignment_id = request.vars.assignment_id
```

Comment:
I don't remember my web2py, but what if `assignment_id` isn't defined? Will the code break?

We really need to get Selenium-based unit tests going.

Reply:
No, it will not break